src/pages/ultimate.tsx

```diff
@@ -0,0 +1,256 @@
+  handleClick: (i: number, j: number) => void;
+}
+
+const boardStyle = {
+  display: 'grid',
+  gridTemplateColumns: 'repeat(3, minmax(auto, max-content))',
+};
+
+export const LargeBoard = (props: LargeBoardProps) => (
+  <div
+    className={css({
+      backgroundColor: 'black',
+      border: '1px solid black',
+      gridGap: 1,
+      ...boardStyle,
+    })}
+  >
+    {props.boards.map((board: Board, i: number) => {
+      const availableBoard = props.available.includes(i);
+      return (
+        <SmallBoard
+          key={i}
+          available={availableBoard}
+          board={board}
+          handleClick={j => {
+            if (availableBoard) {
+              props.handleClick(i, j);
+            }
+          }}
+        />
+      );
+    })}
+  </div>
+);
+
+export interface SmallBoardProps {
+  available: boolean;
+  board: Board;
+  handleClick: (i: number) => void;
+}
+
+export const SmallBoard = (props: SmallBoardProps) => (
+  <div
+    className={css({
+      backgroundColor: props.available ? 'green' : 'red',
+      padding: 10,
+      position: 'relative',
+      ...boardStyle,
+    })}
+  >
+    {calculateWinner(props.board.cells) !== null && (
+      <div
+        className={css({
+          height: '100%',
+          left: 0,
+          position: 'absolute',
+          top: 0,
+          width: '100%',
+        })}
+      >
+        {getSymbol(calculateWinner(props.board.cells))()}
+      </div>
+    )}
+    {props.board.cells.map((cell: Cell, i: number) => (
+      <Square
+        key={i}
+        cell={cell}
+        onClick={props.available ? () => props.handleClick(i) : null}
+      />
+    ))}
+  </div>
+);
+
+export interface SquareProps {
+  cell: Cell;
+  onClick: () => void;
+}
+
+export const Square = (props: SquareProps) => (
+  <button
+    className={css({
+      backgroundColor: 'white',
+      border: '1px solid #999',
+      height: 34,
+      marginRight: -1,
+      marginTop: -1,
+      padding: 0,
+      width: 34,
+    })}
+    onClick={props.onClick}
+  >
+    {getSymbol(props.cell.player)()}
+  </button>
+);
+
+function getSymbol(player: Winner): () => JSX.Element {
+  switch (player) {
+    case Winner.Nought:
+      return () => <Nought />;
+    case Winner.Cross:
+      return () => <Cross />;
+    default:
+      return () => null;
+  }
+}
+
+export const Nought = () => (
+  <svg version="1.1" baseProfile="full" width="100%" height="100%">
+    <circle
+      cx="50%"
+      cy="50%"
+      r="30%"
+      stroke="black"
+      strokeWidth="10%"
+      fill="transparent"
+    />
+  </svg>
+);
+
+export const Cross = () => (
+  <svg version="1.1" baseProfile="full" width="100%" height="100%">
+    <line
+      x1="20%"
+      y1="20%"
+      x2="80%"
+      y2="80%"
+      stroke="black"
+      strokeWidth="10%"
+    />
+    <line
+      x1="20%"
+      y1="80%"
+      x2="80%"
+      y2="20%"
+      stroke="black"
+      strokeWidth="10%"
+    />
+  </svg>
+);
+
+export default Game;

```

Comment:
This file is quite long; I'd be inclined to make a subfolder in components for it and extract some of the components there.